Draft - Images in Prompts (#182)
vellum-ai/vellum-client-generator: fern/docs/content/help/prompts/working-with-images-in-vellum.mdx

---
title: Leverage Images in Your Vellum Prompts and Workflows
description: Learn how to send images to multimodal models like GPT-4 Turbo with Vision from within Vellum’s UI
---

Leverage the power of multimodal models to process both natural language and visual inputs within your LLM-applications using Vellum.

Vellum supports images for OpenAI’s vision models like GPT-4 Turbo with Vision, or `gpt-4-vision-preview` - both via [API](https://docs.vellum.ai/api-reference/) and in the Vellum UI.

> Here’s how to get started using images in Vellum!
> 
> ![Images in Vellum UI](https://storage.googleapis.com/vellum-public/help-docs/images-in-prompts-walkthrough.gif)

Reviewer: nit: Upon initial read, I thought I was supposed to watch the gif to learn how to use images, but then realized that it's just a teaser.

```suggestion

![Images in Vellum UI](https://storage.googleapis.com/vellum-public/help-docs/images-in-prompts-walkthrough.gif)

Read on to learn how to get started using images in Vellum!
```

## Using Images in the UI

Vellum supports images as inputs to both your Prompts and Workflows. In either Sandbox, you can add images inside of a scenario Chat History message.

> Begin by selecting the correct model, `GPT-4 Turbo with vision`, in your Prompt or in a Workflow sandbox Prompt Node.

Reviewer: nit: Prefer to title case Vellum vocab like "Workflow Sandbox".

```suggestion
Begin by selecting the correct model, `GPT-4 Turbo with vision`, in your Prompt or in a Prompt Node within a Workflow Sandbox.
```

![Vision Model Selection](https://storage.googleapis.com/vellum-public/help-docs/images-model-selection.png)

Next, to add an image, drag and drop a valid image into a scenario Chat History message. This converts the Chat Message into an array that can contain multiple image and/or text items.

> <Callout intent="info">
>   Valid image URLs: Images must have their absolute path including the image
>   filetype in their URL (example:
>   https://storage.googleapis.com/vellum-public/help-docs/release-tags-on-deploy.png)
> </Callout>

Reviewer: ```suggestion
<Callout intent="info">
  Valid image URLs: Images must have their absolute path including the image
  filetype in their URL and must be publicly visible (example:
  https://storage.googleapis.com/vellum-public/help-docs/release-tags-on-deploy.png)
</Callout>
```

Here’s what this looks like in the **Prompt Sandbox**:

![Prompt Sandbox Steps](https://storage.googleapis.com/vellum-public/help-docs/images-prompt-steps.png)

> ![Images in Prompt Scenarios](https://storage.googleapis.com/vellum-public/help-docs/images-in-prompts.png)

Reviewer: nit: Upon first read, it's not clear to me what I should be looking for in this image. A purple callout, text above the image, or image caption might help.

And in the **Workflow Sandbox**:

![Workflow Sandbox Steps](https://storage.googleapis.com/vellum-public/help-docs/images-workflow-steps.png)

![Images in Workflow Scenarios](https://storage.googleapis.com/vellum-public/help-docs/images-in-workflows.png)

Once you’ve added in your image, you can configure its settings by clicking the small gear icon to the right of the image. Here you'll be able to adjust things like the `Image Detail` which can have a big impact on token usage (more on that below).

![Image Configuration Steps](https://storage.googleapis.com/vellum-public/help-docs/images-config.png)

You can also switch out an image you’ve dragged in for a new one by updating the image URL in the settings.

![Image Configuration Modal](https://storage.googleapis.com/vellum-public/help-docs/images-config-modal.png)

> ##Image Specifications

Reviewer: Space required to render as an h2

```suggestion
## Image Specifications
```

Here are some important model specifications for GPT-4 Turbo with Vision to keep in mind as you’re incorporating images into your Prompts and Workflows:

- **Number of Images:** No set limit

  There is no fixed number here but token and image size restrictions still apply to determine the number of images that can be sent

- **Image Size:** Less than 20MB

  For prompts and workflows with multiple images, the combined image size should not exceed this limit

- **Supported Image Formats:**

  - JPEG (.jpeg / .jpg)
  - PNG (.png)
  - Non-animated GIF (.gif)
  - WEBP (.webp)

- **Other Notes:**

  - GPT-4 Turbo with Vision does not currently support tool calls so be sure there are no function blocks in your `$chat_history` messages

>   - The Vellum UI currently supports only publicly hosted image urls. To send a base64 image file, you can use Vellum's API instead.

Author: Want to sanity check this and make sure I'm not lying here
